README.md

```diff
@@ -29,23 +29,8 @@ stage = USBXYZStage('COM4')
 ## Usage
 Check the [examples directory](examples) for more usage examples.
 
```

Comment:
This PR explicitly removes communication with M3 USB Devices through Windows as a COM port interface if the hardware was provisioned as such. I propose that we don't take away functionality but instead add to what's already there.

Generally, it looks like the M3 USB Devices can communicate through Windows through one of two drivers either (1) as a native usb device with a custom VID/PID besides the reserved ones or (2) as a COM port using a specific VID/PID.

I'd propose that we either (a) make two interfaces or (b) modify the M3USBInterface to communicate with these devices using the appropriate driver depending on parameters passed into the constructor. (I think choice b is better because it removes some of the responsibility of the user to know which classes to use.)

Here's an example of what that might look like:
````python
class USBInterface(HardwareInterface):
    def __init__(self, com_port: str = None, serial_number: str = None):
        # If the com_port is defined, communicate with the device as a COM port.
        # if the serial_number is defined, communicate with the device via the USBXpress.py library.
        # Error if both or neither parameter is specified.
        ...
````

Reply:
Ok, second benefit to the approach above: you can hide all of the `NewScaleSerial.get_instances()` calls and stick them inside this class. I know you sacrifice the ability to "init the only instance" without knowing the serial number if you plug it in over USB. But that's an easy number to find in a benchtop setup, and a number we should know in advance in a more permanent setup.

Overall, I think device serial number is an ideal way to be able to instantiate this class since it is both fixed and unique across devices. Let's take advantage of what that level of granularity can offer us to hide the `get_instance` calls from the user.